### DISCUSSION TITLE
ESLint rule to enforce `key` property on tags inside `next/head`

### DISCUSSION THREAD

**Discussion by etienne-martin** [vercel/next.js#64624] [Ideas]
### Describe the feature you'd like to request

The [Next.js docs](https://nextjs.org/docs/api-reference/next/head) for the `next/head` component states the following:

> To avoid duplicate tags in your head you can use the key property, which will make sure the tag is only rendered once.

The idea is to enforce this recommendation with an eslint rule to prevent potential bugs (duplicate tags) by spotting cases where a `key` property is missing from tags that live inside `next/head`.

### Describe the solution you'd like

We would need to create new rule for `eslint-plugin-next`. I suggest the name `next-head-key`. Not sure if this should be an error or a warning.

The rule would check for the presence of a `key` prop on:
- Direct children of the `Head` element, 
- Children that are wrapped into maximum one level of `<React.Fragment>`
- Children that are wrapped into a top-level JSX array expression
- Ignore the `<title>` element as it already gets deduped automatically

Here's how I would implement something like this:
https://github.com/vercel/next.js/compare/canary...etienne-martin:next.js:eslint-head-key

### Describe alternatives you've considered

The temporary solution I came up with in a different project was to write a global mock for `next/head` that would throw if a `key` property is missing. Not ideal.

**Comment by balazsorban44** [comment]
This might potentially get noisy if there is only a single tag of a type (unnecessary `key`), but could be off by default, maybe.

**Comment by etienne-martin** [comment]
`next/head` deduplicates tags app-wide, assuming each tag have a `key` property set. The idea for this rule is to make people aware of that feature so that they consciously add keys to avoid duplicate tag insertions _across_ components.

**Comment by Enrique-Ambrocio** [comment]
Hi there! Is this issue still open and if so is there any way that I can help? I'm actually new to contributing to open source and looking to tackle my first issue.

**Comment by etienne-martin** [comment]
Hi @Enrique-Ambrocio, the issue is still open and we haven't decided yet if we're going to go forward with my suggestion. However, I see that @puneetkathar1 has already started working on it: https://github.com/vercel/next.js/pull/39975

**Comment by Enrique-Ambrocio** [comment]
I see. No worries! If anything should ever come up please don't hesitate to reach out to me. Also, if you by any chance come across potential opportunities, could you possibly let me know? Thank you! @etienne-martin
